paddle/framework/executor.cc

```diff
@@ -0,0 +1,177 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "paddle/framework/executor.h"
+#include <memory>
+#include "paddle/framework/op_registry.h"
+#include "paddle/framework/operator.h"
+#include "paddle/framework/scope.h"
+#include "paddle/platform/device_context.h"
+
+namespace paddle {
+namespace framework {
+
+class LinearListView;
+class GraphView;
+
+// Immutable view of a ProgramDesc organized for efficient execution.
+class ProgramDescView {
+ public:
+  virtual ~ProgramDescView() {}
+  virtual void Initialize(const ProgramDesc*) = 0;
+  static ProgramDescView* Create(bool is_linear);
+};
+
+class LinearListView : public ProgramDescView {
+ public:
+  void Initialize(const ProgramDesc*) override;
+
+ private:
+  std::vector<std::unique_ptr<OperatorBase>> ops_;
+};
+
+class GraphView : public ProgramDescView {
+ public:
+  void Initialize(const ProgramDesc*) override;
+};
+
+ProgramDescView* ProgramDescView::Create(bool is_linear) {
+  if (is_linear) {
+    return new LinearListView();
```

Comment:
Why do we need `LinearListView` since we already have `GraphView`?

Reply:
- `LinearListView` organizes the topology in a linearlist, and operators will be executed sequentially. 
- `GraphView` organizes the topology in a Graph, and further optimization can be applied based on Graph structure.
I think that we can have `LinearListView` at now, maybe `GraphView` can be implemented later.